src/InterfaceUserPref.py

```diff
@@ -2,24 +2,24 @@
 To provide a set group of methods for other classes
 to access the UserPreferences Class
 """
+import pathlib
 from abc import ABC, abstractmethod
 
 class InterfaceUserPref(ABC):
     """
-    Abstract class to provide methods the to define
-    the accessible methods of the UserPreferences Class
+    Abstract class to define methods to
+    access the UserPreferences Class
     """
     @abstractmethod
-    def check_if_path_exists(self) -> bool:
-        """ Check if there exists a saved path """
-        pass
-
-    @abstractmethod
-    def save_default_path(self, path:str) -> None:
+    def save_default_path(self, path:pathlib.Path) -> None:
```

Comment:
This is fine, but that leaves the caller to determine if this was successful, and the implementer to raise an exception or error if it isn't.  So you might want to document that (both for any future callers and for any future implementers).